File: spec/support/factories/employment_record.rb
# Read about factories at https://github.com/thoughtbot/factory_girl
FactoryGirl.define do
  factory :employment_record do
    candidate
    company_name { Faker::Company.name }
    job_title { Faker::Job.title }
    description { Faker::Lorem.sentences(4) }
    start_date  { 18.years.ago }
    end_date    { Date.today - 1.year }
    city { Faker::Address.city }
    country_iso3 { Carmen::Country.all[rand(0..(Carmen::Country.all.size-1))].alpha_3_code }
  end
end

File: app/assets/javascripts/app/features/candidate/resume/employment/models/employment-record.js
'use strict';

angular.module('bridge')

  .run(['modelize', function (modelize) {
    modelize.defineModel('employmentRecord', {
      baseUrl: '/candidate/employment-records',

      companyName: '',
      jobTitle: '',
      country: modelize.attr.model({ modelClass: 'country' }),
      countryIso3: '',
      city: null,
      description: '',
      startDate: null,
      endDate: null,
      isCurrentJob: false
    });
  }])

  .factory('EmploymentRecord', ['modelize', function (modelize) {
    return modelize('employmentRecord').$modelClass;
  }]);
